src/Kunstmaan/MediaBundle/Helper/MediaManager.php

```diff
@@ -160,4 +219,26 @@ public function getFolderAddActions()
 
         return $result;
     }
+
+    /**
+     * @throws Exception
+     */
+    public function isExtensionAllowed(File $file): bool
+    {
+        $mimeType = $this->mimeTypes->guessMimeType($file->getRealPath());
+        $extension = $this->mimeTypes->getExtensions($mimeType)[0];
+        if (!in_array($extension, self::ALLOWED_EXTENSIONS, true)) {
+            return false;
+        }
+
+        if (in_array($extension, self::IMAGE_EXTENSIONS, true)) {
+            $fileUploadModel = new ImageUploadModel($file);
```

Comment:
I'm not sure if we need a whole object to just check if a file is corrupt or not, we could perform the check behind the `detectCorrupted` flag manually here. But no strong opinion on this, so others can decide what to do here.

If we keep the object I would unset it after the check just to avoid memory issues on large bulk uploads

Reply:
I chose this way because when symfony improves the Assert\Image annotation we would automatically benefit from the improvements without having to adjust any code. If there are better ways to achieve this I am happy to rewrite it.